src/nnet3/nnet-common.cc

```diff
@@ -412,6 +413,12 @@ size_t IndexVectorHasher::operator () (
     ans += iter->n * 1619;
     ans += iter->t * 15649;
     ans += iter->x * 89809;
+	// check 'n2 > index_vector.size()' first,
+	// otherwise, 'end - n2' will be less than 'index_vector.begin()' and
+	// cause error "vector iterator + offset out of range" in STL vector.
```

Comment:
Please make sure this code is correctly indented (and make sure you use spaces not tabs).
Also, you can change the comment to the following:
  // The following if-statement was introduced in order to fix an
  // out-of-range iterator problem on Windows.